pgjdbc/src/main/java/org/postgresql/jdbc/PgResultSet.java

```diff
@@ -734,6 +734,30 @@ public int getConcurrency() throws SQLException {
     return getTimestampUtils().toLocalDateTime(string);
   }
 
+  private @Nullable LocalDate getLocalDate(int i) throws SQLException {
+    byte[] value = getRawValue(i);
+    if (value == null) {
+      return null;
+    }
+
+    if (isBinary(i)) {
+      int col = i - 1;
+      int oid = fields[col].getOID();
+      if (oid == Oid.DATE) {
+        return getTimestampUtils().toLocalDateBin(value);
+      } else if (oid == Oid.TIMESTAMP) {
+        return getTimestampUtils().toLocalDateTimeBin(value).toLocalDate();
+      } else {
+        throw new PSQLException(
+            GT.tr("Cannot convert the column of type {0} to requested type {1}.",
+                Oid.toString(oid), "date"),
+            PSQLState.DATA_TYPE_MISMATCH);
+      }
+    }
+
+    return getTimestampUtils().toLocalDateTime(castNonNull(getString(i))).toLocalDate();
```

Comment:
I wonder if we want to gradually remove `getTimestampUtils()`.
For instance, `byte[] -> LocalDate` conversion does not really need `TimestampUtils` instance. Should the method be static instead?

Then it would ease understanding of what `toLocalDateTime` could use, and it would avoid accidental use of shared state. WDYT?

Reply:
I was thinking about that, too. Theoretically the TimestampUtils should only contain the legacy stuff. The java.time stuff could completely be moved to PgResultSet.
As first step I'd make the methods static, as suggested.

Reply:
I would do this in a separate cleanup PR. I made it consistent with the LocalDateTime code.